XO.py

```diff
@@ -0,0 +1,157 @@
+from random import randint
+from collections import namedtuple
+
+# Size of board.
+N = 3
+
+
+class Model:
+    def __init__(self):
+        self.board = [[None for x in range(N)] for y in range(N)]
+
+    def get_board_status(self):
+        return self.board
+
+    def update_board(self, line, col, players, curr_player):
+        if self.board[line][col] is not None:
+            return False
+        else:
+            self.board[line][col] = players[curr_player].symbol
+            return True
+
+    def line_won(self, players, curr_player):
+        for line in self.board:
+            if all(x == players[curr_player].symbol for x in line):
+                return True
+        return False
+
+    def col_won(self, players, curr_player):
+        for col in range(N):
+            if all(line[col] == players[curr_player].symbol for line in self.board):
+                return True
+        return False
+
+    def diagonal_won(self, players, curr_player):
+        # Main diagonal.
+        if all(self.board[index][index] == players[curr_player].symbol for index in range(N)):
+            return True
+        # Opposite diagonal.
+        if all(self.board[index][N-1-index] == players[curr_player].symbol for index in range(N)):
+            return True
+        return False
+
+
+class View:
+    def __init__(self):
+        pass
+
+    def print_board(self, board):
+        print()
+        print('Board:')
+        for line in board:
+            for item in line:
+                if item is None:
+                    print('-', end='')
+                else:
+                    print(item, end='')
+            print()
+        print()
+
+    def declare_winner(self, players, curr_player):
+        print()
+        print('***')
+        print(f'The winner is {players[curr_player].name}!!!')
+        print('***')
+
+    def declare_tie(self):
+        print()
+        print('***')
+        print('The result is a tie!!!')
+        print('***')
+
+
+class Controller:
+    def __init__(self, model, view):
+        self.model = model
+        self.view = view
+        self.players, self.curr_player = self.initiate_players()
+        self.number_of_moves_left = N * N  # Equals to number of squares on board.
+
+    def initiate_players(self):
+        Player = namedtuple('Player', 'name symbol')

```

Comment:
named tuples are a nice idea here, but they're usually treated like classes.
This line should move to the top of the file, outside of any classes.